R/lockfile.R

```diff
@@ -193,7 +193,13 @@ renv_lockfile_create_impl <- function(project, libpaths, type, packages, exclude
 
   lockfile <- renv_lockfile_init(project)
 
-  records <- renv_snapshot_libpaths(libpaths = libpaths, project = project)
+  packages <- packages %||% renv_snapshot_dependencies(project, type)
+
+  records <- renv_snapshot_packages(
```

Comment:
This is really the meat of the PR; instead of snapshotting the whole library paths and then filtering, we just snapshot the specific packages we need.